### PR TITLE
#1841 - sgcollect: Collect sg binary from user-specified path

### PR DESCRIPTION
**Pull request by mattcarabine**
This change is [Reviewable](https://reviewable.io/reviews/couchbase/sync_gateway/1842)

### CODE REVIEW

**Comment by mattcarabine** [comment]
Resolves #1841

File: tools/sgcollect_info [hunk 1]
```diff
@@ -454,6 +454,11 @@ def main():
 
     # Collect the SG binary
     sg_binary_path, _ = get_paths_from_expvars(sg_url)
+    if options.sync_gateway_executable_path is not None and len(options.sync_gateway_executable_path) > 0:
+        if not os.path.exists(sync_gateway_executable_path):
```
Comment (tleyden) [inline tools/sgcollect_info:458]: if options.sync_gateway_executable_path is None or has zero length, can you print a log warning of some sort like "No sync_gateway_executable_path specified, so not collecting sync gateway binary"?  I think that will help debug the cases when the zip is missing an sg binary and we are wondering why ..

**Comment by mattcarabine** [comment]
Well we already log if we failed to collect a binary (and the path that we tried to fetch it from), either from the exp_var binary path or the user supplied one (https://github.com/couchbase/sync_gateway/blob/master/tools/sgcollect_info#L461). 
I suppose I can add a more specific one saying that we couldn't find a binary path from exp_vars (if that's possible) and there was no user supplied path either.

Is that the kind of thing you were looking for?

If not please could you clarify what you're asking for logging-wise?

File: tools/sgcollect_info (hunk 1 above)
Comment (mattcarabine) [inline tools/sgcollect_info:458]: Reposting as I put the comment in the wrong place.

Well we already log if we failed to collect a binary (and the path that we tried to fetch it from), either from the exp_var binary path or the user supplied one (https://github.com/couchbase/sync_gateway/blob/master/tools/sgcollect_info#L461). 
I suppose I can add a more specific one saying that we couldn't find a binary path from exp_vars (if that's possible) and there was no user supplied path either.

Is that the kind of thing you were looking for?

If not please could you clarify what you're asking for logging-wise?

Comment (tleyden) [inline tools/sgcollect_info:458]: > Well we already log if we failed to collect a binary (and the path that we tried to fetch it from), either from the exp_var binary path or the user supplied one 

Ok that's good enough.  Will merge.

**Pull request merged**